Content.Shared/Prototypes/LocalizedPrototype.cs

```diff
@@ -0,0 +1,37 @@
+using System.Linq;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Prototypes;
+
+public abstract class LocalizedPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; } = default!;
+
+    public const string LocFormat = "{type}-{ID}-{field}";
+
+    /// <summary>The localization string for the name of this prototype</summary>
+    public string NameLoc => ToLocalizationString("name");
+    /// <summary>The localized string for the name of prototype</summary>
+    public string Name => Loc.GetString(NameLoc);
```

Comment:
```suggestion
    public string Name => Loc.TryGetString(NameLoc) ?? Name;
    
    [DataField]
    private string _name = "";
```

Reply:
I do not like this.

Reply:
This opens up major flexibility for downstreams who may not want to do everything 'properly'.
It's quite a lot more effort to make individual FTL files for every single Prototype made and is only useful at large scale when planning for people who may fork and translate. Even then it still doesn't benefit *you*, only your potential downstreams. It may not matter for EE if the plan is to *require* the use of FTL files but that's still something we can just enforce while ensuring our systems are as user friendly as possible to people who don't care.